### ISSUE TITLE
False positive for the Java/Env/Process

### ISSUE THREAD

**Issue by uturunku1** [newrelic/newrelic-diagnostics-cli#28]
[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

## Description

Customers will see the following nrdiag result summary when their newrelic.jar has been renamed to, say, newrelic-4.9.0.jar to indicate a specific version:

```
Failure - Java/Env/Process
None of the active Java processes included the -javaagent argument. For proper installation of New Relic Java agent, the -javaagent flag must be passed to the same Java process that is running your application
```
That is because our regex search explicitly looks for "newrelic.jar" and is not matching the actual "newrelic-someversionnumber.jar" on the customers' system.

## Where to place that fix
https://github.com/newrelic/newrelic-diagnostics-cli/blob/main/tasks/java/env/process.go#L118

**Comment by MikeBLambert** [comment]
Hi! 
Questions about the requirements surrounding this bug: 

Is it ok for the user to rename the file to anything? (i.e. bananarelik-superfun-random-filename-also-i-like-spaghetti-and-am-including-that-in-my-filename-just-cuz-anyways-yeah-now-im-gonna-mash-the-keypad-lsdksf-1.8.0.jar) or are we assuming that the filename will still have 'newrelic' in it and just have a version suffixed onto the end?

If it's the former, do we want to deal with it via an optional command line argument or do you think that there is a better way to go about it?

**Comment by uturunku1** [comment]
Hey @MikeBLambert! Great question there. I was considering the use case where customers are not manually renaming the newrelic.jar, but instead, they are downloading a newrelic.jar that got rename into newrelic[versionnumber].jar by the Java agent's pipeline when it releases a new version. However, you are right that it is possible that customers rename it to something completely different, following their own naming conventions, that does not even match the word "newrelic". All we can be certain about is that at least they will keep the extension ".jar" part of the name intact. In which case, your suggestion for regex `regex := regexp.MustCompile("-javaagent:(.+).jar")` should work because it allows for that flexibility of renaming the jar.
I'd say yes go ahead with that regex rule, though not being an expert in the Java agent, I wonder if allowing for this broad match could cause an issue down the road. In other words, if our Java task succeeds after finding that the customer passed `-javaagent:mypath/bananarelik-superfun-random-filename-also-i-like-spaghetti-and-am-including-that-in-my-filename-just-cuz-anyways-yeah-now-im-gonna-mash-the-keypad-lsdksf-1.8.0.jar`, are we hiding a possible issue with the java agent configuration and misleading the customer with that success status?
@MellowMarble can we have your perspective on this? Do you see any issue with `regexp.MustCompile("-javaagent:(.+).jar")` ?

**Comment by MellowMarble** [comment]
Oh that's a tricky one. Technically the Agent could have any name ending in .jar but if we looked for that we could end up picking up other, non-Relic APM Agents as well. The versioned newrelic-XXX.jar name seems like it'd catch most of the edge-cases though. I don't think I've ever seen a customer using our JAR with a name that didn't include "newrelic" somewhere in it. 

If we were to look for anything ending in .jar we would probably want to be able to run some code directly against the JAR that gets found to ensure it includes New Relic classes somehow; I'm not sure what's available in Go for that, however.

**Comment by MikeBLambert** [comment]
Thanks @MellowMarble and @uturunku1! If it's ok, I'll go ahead and open up a PR with the fix. I'm very excited because this is potentially going to be my very first open source contribution! 🤞

**Comment by uturunku1** [comment]
Issue resolved 🎉 by this PR: https://github.com/newrelic/newrelic-diagnostics-cli/pull/40
